nc-news-fe/src/components/SelectedArticle.jsx

```diff
@@ -39,10 +90,29 @@ function SelectedArticle() {
       <p>Related to {article.topic}</p>
       <img src={article.article_img_url}></img>
       <p className="article-body">{article.body}</p>
-      <p>{article.votes} votes</p>
-      <Link to="/articles">
-        <button>Back To Articles</button>
-      </Link>
+      <div className="votes-counter">
+      <p>{votes} votes</p>
+      </div>
+      <div className="voting-buttons">
+        <button
+          className="upvote-btn"
+          onClick={handleUpvote}
+          disabled={hasUpvoted}>
+          Upvote
+        </button>
+        <button
+          className="downvote-btn"
+          onClick={handleDownvote}
+          disabled={hasDownVoted}>
+          Downvote
+        </button>
```

Comment:
**When I throttle my internet to be slow**
I currently can spam the vote button and your frontend lets those votes go through. We want the vote button to be disabled as soon as someone votes to stop this from happening if their internet happens to be slow. 

**When I go offline**
Your error message shows which is fab but I also want the vote buttons to be disabled and for the vote number to ideally not change (or go back to the original vote that was last fetched). Test it out yourself and you should see what I'm looking at.

Reply:
Additionally I ideally want to be able to vote in either direction. E.g. I should be able to downvote, then upvote back to the original number when I went onto the site, and then upvote by one in the other direction. So the net change is one vote but I can change my mind on which direction i vote in even if I still only get one vote.